src/ray/raylet/node_manager.cc

```diff
@@ -1421,9 +1424,31 @@ void NodeManager::TreatTaskAsFailed(const Task &task, const ErrorType &error_typ
     // information about the TaskSpecification implementation.
     num_returns -= 1;
   }
+  // Determine which IDs should be marked as failed.
+  std::vector<plasma::ObjectID> objects_to_fail;
+  if (num_returns == 0 && required_object_id != nullptr) {
+    objects_to_fail.push_back(required_object_id->ToPlasmaId());
```

Comment:
I found this `if` condition a little confusing...can you add a comment to explain why we use `required_object_id` in this case?